prometheus-ksonnet/lib/prometheus.libsonnet

```diff
@@ -52,7 +52,8 @@
       container.withArgs([
         '-v',
         '-t',
-        '-p=/etc/prometheus',
+        // Watch the directory in which the Prometheus config file is located.
+        '-p=' + std.substr(_config.prometheus_config_file, 0, std.reverse(std.findSubstr('/', _config.prometheus_config_file))[0]),
```

Comment:
you could do this more easily. What you want is the config dir. Do this in `lib/config.libsonnet`:
```
-    prometheus_config_file: '/etc/prometheus/prometheus.yml',
+    prometheus_config_dir: '/etc/prometheus',
+    prometheus_config_file: self.prometheus_config_dir + '/prometheus.yml',
```
Then you can refer to `_config.prometheus_config_dir` instead, and then, in `prometheus-ha-mixin.libsonnet` set `prometheus_config_dir` instead of `prometheus_config_file`. Reasonable?